Makefile

```diff
@@ -1,103 +1,133 @@
+.PHONY: help install-python install-js install-test install \
+		update-npm-requirements javascript sass \
+		extract_translations compile_translations generate_dummy_translations validate_translations \
+		detect_changed_source_translations pull_translations push_translations check_translations_up_to_date \
+		quality test-python render-templates test-js test-js-debug test test-acceptance test-a11y test-sandbox \
+		install-osx-requirements
+
+.DEFAULT_GOAL := help
+
+help: ## display this help message
+	@echo "Please use \`make <target>' where <target> is one of"
+	@perl -nle'print $& if m{^[a-zA-Z_-]+:.*?## .*$$}' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m  %-25s\033[0m %s\n", $$1, $$2}'
+
+clean: ## remove generated byte code, coverage reports, and build artifacts
+	find . -name '__pycache__' -exec rm -rf {} +
+	find . -name '*.pyc' -exec rm -f {} +
+	find . -name '*.pyo' -exec rm -f {} +
+	find . -name '*~' -exec rm -f {} +
+	coverage erase
+	rm -fr build/
+	rm -fr dist/
+	rm -fr *.egg-info
+
 ##################
 # Install commands
 ##################
-install-python:
+
+install-python: ## Install python dependencies
 	pip install -r requirements/django.txt
 	pip install -r requirements/base.txt --only-binary=lxml,libsass
 
-install-js:
+install-js: ## install JavaScript dependencies
 	npm install
 
-install-test:
+install-test: ## install requirements for tests
 	pip install -r requirements/test.txt
 
-install: install-python install-js install-test javascript sass
+install: install-python install-js install-test javascript sass ## install all dependencies
+
+upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
+upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+	pip install -q pip-tools
```

Comment:
I recommend getting meta and using pip-tools to version pip-tools. Surprisingly, different versions of pip-tools will produce different `.txt` files sometimes even install the pins differently.

Example: https://github.com/edx/registrar/blob/master/Makefile#L37-L46